update_db.py

```diff
@@ -0,0 +1,197 @@
+import base64
+import json
+import re
+from apiclient import errors
+from collections import namedtuple
+
+import app
+from app.lib.gmail_api import GmailApi
+from app.models import User, Message
+from config.secrets import USER_INFO_BY_EMAIL, GMAIL_QUERIES
+
+
+# TODO don't run query up here?
+EMAIL_TO_USER_ID = {user.email: user.id for user in User.query.all()}
+
+db = app.db
+service = GmailApi().get_service()
+
+MessageData = namedtuple('MessageData', [
+    'body',
+    'timestamp',
+    'message_id',
+    'label',
+    'data',
+    'sender_user_id',
+    'thread_id',
+])
+
+
+def create_or_update_users():
+    for email, user_attributes in USER_INFO_BY_EMAIL.items():
+        existing_user = User.query.filter_by(email=email).first()
+        if existing_user is not None:
+            print("Updating: {}".format(email))
+            User.query.filter_by(email=email).update(user_attributes)
+        else:
+            print("Creating: {}".format(email))
+            new_user = User(**user_attributes)
+            db.session.add(new_user)
+
+    db.session.commit()
+
+
+def fetch_and_save_new_messages(query):
+    """
+    Get all messages from gmail api and save to db. This happens in a few steps - 
+    1. use gmail api messages.list() to get a list of all message ids returned from query
+    2. loop through each of those messages ids and use messages.get() to get more detailed info
+    3. parse through response so we can get get the fields we want to save in the db
+
+    Params
+        query [String] passed to gmailAPI to fetch messages
+    Returns
+        [Int] Count of new messages saved
+    """
+    existing_message_ids = set(m.message_id for m in Message.query.all())
+
+    try:
+        print("Fetching new messages from query: {}".format(query))
+        response = service.users().messages().list(userId='me', q=query).execute()
```

Comment:
I was looking at https://developers.google.com/gmail/api/guides/sync#full_synchronization, and maybe there's an opportunity to store the `historyId`, so that subsequent queries can pass in the last message scraped and get only new messages (instead of fetching the full list and deduping against `existing_message_ids`).

May not be worth it, depending on how expensive the searching part of the script is.

Reply:
yeah, I was thinking another way of doing this is just getting the timestamp of the most recently stored message and then querying for messages after that one. Definitely don't want  to fetch all of them all over again (there are so many!)

Reply:
Looks the historyId is used if you're trying to get all updates to your inbox. You can't search specific queries. Also, `History records are typically available for at least one week and often longer. However, the time period for which records are available may be significantly less and records may sometimes be unavailable in rare cases.`

I'm gonna try to get the most recent messages based on the timestamp from our most recent message, but I think we may end up missing some messages that way :(